### PR TITLE
add docker tutorial for core & nlu #1239

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- see https://github.com/RasaHQ/rasa_core/issues/1239

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] add steps to set up core
- [x] add steps to set up core & nlu
- [x] instructions for custom pipeline
- [x] add steps for a custom action server
- [ ] add steps for duckling
- [x] add steps for custom tracker stores
- [x] updated the changelog

### CODE REVIEW

File: docs/docker_walkthrough.rst [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+    docker -v && docker-compose -v
+    # Docker version 18.06.1-ce, build e68fc7a
+    # docker-compose version 1.22.0, build f46880f
+
+If your output is not similar to the one above, please install Docker.
+See `this instruction page <https://docs.docker.com/install/>`_
+of the Docker documentation which describes how to install Docker on a variety
+of operating systems.
+
+Congratulation! 🚀 Now we have got everything set up to get going!
```
Comment (JustinaPetr) [inline docs/docker_walkthrough.rst]: The same with using 'we'. I would use 'you' everywhere because it refers to the user directly.  One more thing to this is that I would probably leave the 'Congratulations' part here because it makes it look like the most important part of the tutorial is already finished  :)

File: docs/docker_walkthrough.rst [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+    - create some story lines for our chatbot
+    - train our chatbot
+    - run our chatbot within a Docker container
+
+2.1 Create Stories
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Rasa Core works by learning from example conversations. Therefore, we now need
+to write a few examples to get started.
+Stories are written in `Markdown <https://en.wikipedia.org/wiki/Markdown>`_
```
Comment (JustinaPetr) [inline docs/docker_walkthrough.rst]: I would suggest having this part after you explain which directories a user will have to create because there is too much going back and forth. Maybe it would even make sense to create a separate step for preparing the directory and then diving into stories

File: docs/docker_walkthrough.rst [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+  docker run \
+    -v $(pwd):/app/project \
+    -v $(pwd)/models/rasa_core:/app/models \
+    rasa/rasa_core:latest \
+    train \
+      --domain project/domain.yml \
+      --stories project/data/stories.md \
+      --out models
+
+Yey! You just trained your chatbot for the first time!
```
Comment (JustinaPetr) [inline docs/docker_walkthrough.rst]: I think there are too many excitement words included :) For documentation, I would suggest keeping documentation content a bit more toned down because developers should use the docs as a concise and guide of how to use the software.

Comment (JustinaPetr) [reply docs/docker_walkthrough.rst]: Also, I think it's better to leave out 'for the first time' part, because the quickstart is the place where we expect people to build their first Rasa assistant.

File: docs/docker_walkthrough.rst [hunk 4]
```diff
@@ -0,0 +1,539 @@
...
+  ]
+
+If the bot cannot understand you (which is quite like since we provided only
+very little NLU training data), the answer is something along the lines:
+
+.. code-block:: bash
+
+  []
+
+Hooray! 🎉 🎊 We can now train Rasa Core, Rasa NLU and speak to our bot in
```
Comment (JustinaPetr) [inline docs/docker_walkthrough.rst]: Should this really be about training? I think the training steps are already included above.

**Comment by wochinge** [comment]
@JustinaPetr I implemented your comments and added the next steps.
It is still quite long, but I don't know where to shorten it without losing information.
I am also not sure about Duckling since I don't know how common it is to use Duckling as part of the NLU pipeline. Generally I tend to include in the tutorial as then everything is covered.

File: docs/docker_walkthrough.rst [hunk 5]
```diff
@@ -0,0 +1,684 @@
...
+      command:
+        - start
+        - --path
+        - models
+        - -c
+        - config/nlu_config.yml
+
+.. note::
+
+  Depending on the selected pipeline you might have to use a different
```
Comment (JustinaPetr) [inline docs/docker_walkthrough.rst]: Where a user can find more information about that? Would be good to add a link or something, because people do you their own custom configs very often.

File: docs/docker_walkthrough.rst [hunk 6]
```diff
@@ -0,0 +1,684 @@
...
+    def name(self):
+      return "action_joke"
+
+    def run(self, dispatcher, tracker, domain):
+      request = requests.get('http://api.icndb.com/jokes/random').json() #make an api call
+      joke = request['value']['joke'] #extract a joke from returned json response
+      dispatcher.utter_message(joke) #send the message back to the user
+      return []
+
+Then add the custom action in your stories and your domain file.
```
Comment (JustinaPetr) [inline docs/docker_walkthrough.rst]: 'Then' repeats two times in a row. What about alternating it with 'Next'? :)

File: docs/docker_walkthrough.rst [hunk 7]
```diff
@@ -0,0 +1,684 @@
...
+Continuing the example from above replace replace ``utter_cheer_up`` in
```
Comment (JustinaPetr) [inline docs/docker_walkthrough.rst]: Typo: 'replace' is written two times :)

File: docs/docker_walkthrough.rst [hunk 8]
```diff
@@ -0,0 +1,684 @@
...
+
+As for Rasa NLU, it is necessary to tell Rasa Core the location of the action
+server. Add this to your ``config/endpoints.yml``:
+
+.. code-block:: yaml
+
+  action_endpoint:
+    url: http://action_server:5055/webhook
+
+If you then run ``docker-compose up`` the action server starts together
```
Comment (JustinaPetr) [inline docs/docker_walkthrough.rst]: I would rephrase this as a mandatory step: 'Run 'docker-compose up' to start the action server together with...'.

**Comment by JustinaPetr** [comment]
Awesome! I would be keen on leaving the duckling out just because it's very optional unless there are some special configurations which a developer would have to implement for the duckling to work. Is it very different from adding additional components and adding it to the config files?

**Comment by wochinge** [comment]
@JustinaPetr Adding Duckling is just as the other components, except that the url of has to be added within the NLU pipeline

**Comment by wochinge** [comment]
@JustinaPetr Updated it according to your notes

**Comment by JustinaPetr** [comment]
@wochinge Looks good! :)
